### DISCUSSION TITLE
Revision Series Plotting to look at Forecast Consistency

### DISCUSSION THREAD

**Discussion by hertneky** [dtcenter/METplus#1306] [Plot Generation] [answered]
I am looking at this [METplus use-case](https://metplus.readthedocs.io/en/main_v4.0/generated/model_applications/precipitation/MTD_fcstHRRR-TLE_FcstOnly_RevisionSeries_GRIB.html#sphx-glr-generated-model-applications-precipitation-mtd-fcsthrrr-tle-fcstonly-revisionseries-grib-py) for creating a revision series to look at forecast consistency. I have run this use-case on a single case, and loaded the data into METviewer (MTD>mv_mtd_wpc_fc) but I am not sure how to compute the revision series (looking at an attribute, i.e. area, with decreasing lead). I have also looked into METcalcpy/METplotpy and didn't see anything obvious there either. Note: it was mentioned there was a revision series flag that needed to be set.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @hertneky, I reached out to the METplus development team on Slack, trying to identify the scientist who knows the most about that use case. As I understand it, that use case runs MTD in a "revision series" sort of way. By that I mean that it holds the valid time constant and defines a time series of 13 lead times (hours 0, 1, ... 12) across multiple initialization hours.

MTD processes the time series, generates 3D space/time objects, and writes output that can be loaded into METviewer. I understand that your question stems from this sentence in the Use Case documentation:
"The output can then be loaded into METviewer to compute the revision series and assess the consistency either of one case or many."

Now that you've loaded it into METviewer, how can you actually make that revision series plot. I have never actually made a revision plot myself in METviewer.

@TatianaBurek can you provide some direction here? Do you have an example of revision series plots through METviewer, particularly using MTD output?

**Reply by TatianaBurek** [reply]
@hertneky, you can find some helpful info about the revision series in the comment by Tressa:  https://github.com/dtcenter/METviewer/issues/111. Also there is a Revision History manual.pdf file with an example from Tressa.
I  created an example  plot on dakota: http://dakota.rap.ucar.edu:8080/metviewer_dev/metviewer1.jsp #20211222_115539.
Here is XML and the image
![plot_20211222_115539](https://user-images.githubusercontent.com/21346768/147142148-5d0dad06-27a1-4e36-a938-bc24413bde5d.png)
[plot_20211222_115539.xml.txt](https://github.com/dtcenter/METplus/files/7764664/plot_20211222_115539.xml.txt)
This plot can be built using Python

**Reply by JohnHalleyGotway** [reply]
Hi Tracy, just following up on this Discussion. Does Tatiana's example answer your question? Or do you have additional ones?

If it does answer your question, feel free to click the "Mark As Answer" button to indicate that. We ultimately "Lock conversation" on all answered discussions to encourage users to post new questions as new discussion topics.

**Reply by hertneky** [reply]
@TatianaBurek Thank you for the information. This work is in progress again. After looking through this, I have one follow-up question. Where does the variable REV_PROB(APCP_01...) come from, as I assume that this is the revision of the MTD variable PROB (probabilistic precip estimates), but it is not clear where that revision variable is computed? 

What I currently have is generic (i.e. no revision) MTD output following the revision series use-case config example provided (using different input data).
